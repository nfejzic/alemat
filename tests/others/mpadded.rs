use alemat::{
    attributes::Attribute,
    elements::{Frac, Num, Padded, PaddedAttr},
    MathMl, MathMlFormatter,
};

#[test]
fn padded() {
    // example from https://www.w3.org/TR/mathml-core/#adjust-space-around-content-mpadded

    let out = MathMl::with_content(
        Padded::from(alemat::children![Frac::builder()
            .num(Num::from(23456))
            .denom(Num::from(78))
            .build()])
        .with_attr([
            PaddedAttr::LeftSpace("2em".into()),
            PaddedAttr::VerticalOffset("-1em".into()),
            PaddedAttr::Height("1em".into()),
            PaddedAttr::Depth("3em".into()),
            PaddedAttr::Width("7em".into()),
            PaddedAttr::Global(Attribute::Style("background: lightblue;".into())),
        ]),
    )
    .render_with(&mut MathMlFormatter);

    crate::snap_test!(out, name: "others_padded");
}
