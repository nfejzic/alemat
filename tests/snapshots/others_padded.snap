---
source: tests/others/mpadded.rs
expression: input
---
<math>
  <mpadded lspace="2em" voffset="-1em" height="1em" depth="3em" width="7em" style="background: lightblue;">
    <mfrac>
      <mn>
        23456
      </mn>
      <mn>
        78
      </mn>
    </mfrac>
  </mpadded>
</math>

